src/util/indexrange.h

```diff
@@ -67,58 +67,58 @@ class IndexRange final: private std::pair<SINT, SINT> {
     }
 
     SINT length() const {
-        return (head() <= tail()) ? (tail() - head()) : (head() - tail());
+        return (start() <= end()) ? (end() - start()) : (start() - end());
     }
 
-    // Clamps index by this range including both head() and tail()
+    // Clamps index by this range including both start() and end()
     // boundaries.
     SINT clamp(SINT index) const {
-        if (head() <= tail()) {
-            return std::max(head(), std::min(tail(), index));
+        if (start() <= end()) {
+            return std::max(start(), std::min(end(), index));
         } else {
-            return std::min(head(), std::max(tail(), index));
+            return std::min(start(), std::max(end(), index));
         }
     }
 
     bool contains(SINT index) const {
-        if (head() <= tail()) {
-            return (head() <= index) && (index < tail());
+        if (start() <= end()) {
+            return (start() <= index) && (index < end());
         } else {
-            return (head() >= index) && (index > tail());
+            return (start() >= index) && (index > end());
         }
     }
 
     // Splits this range into two adjacent parts by slicing off
     // and returning a range of given length and same direction
-    // from the head side. The given head length must not exceed
+    // from the front side. The given front length must not exceed
     // the length of this range.
-    IndexRange splitHead(SINT headLength);
+    IndexRange splitFront(SINT frontLength);
```

Comment:
This can be read as "split the Front part". Ideas: cutOutFront() .. cutFront() .. sliceOutFront() .. sliceOffFront()